README.md

```diff
@@ -24,76 +24,276 @@ undirected graphs, and electrical circuits.
+
+		let vertex2string = lam s.
+			int2string s
+		let label2string = lam s.
+			char2string s
+
+The model constructors for the types are:
+
+* DFA 
+
+  `DFA(dfa,input, state2string, label2string,[])`
+* NFA 
+
+  `NFA(nfa,input, state2string, label2string,[])`
+* Digraph : 
+    
+	`Digraph(digraph, vertex2string,label2string,[]),`
+* Graph 
+
+  `Graph(graph, vertex2string,label2string,[])`
+
+
+* BTree   
+
+  `BTree (btree,node2string,[])`
+
+To create the visualizer, use this function:
+
+	visualize data
+
+Where `data` is a list of models.
+
+There is also the option to define a display name for any of the nodes when visualizing any of the datatypes defined above. These values must be strings, and have no affect on the model other than that when visualized the labels for the states will not be the names used in the model, but the display names. All names used in the model must be unique, but display names do not. The display names are defined by a list of touples `(a,b)`, where `a` is the name of the node that is used in the model and `b` is the string that will be shown as the label instead. To use the display names, the list will be added to the model constructor as a argument. For example, if you want to have display names for a DFA, the constructor would be:
+
+  `DFA(dfa, input, state2string, label2string, displayNames)`
+
+Where displayNames is a list of touples containing the nodes that should have different display names and their new labels.
+
+**Note: this does not work with file conversion at the moment. 
+
+# Creating files with the datatypes
+Before you can start converting your models to pdf and other formats, you need to
+install the graphviz package. Follow the instructions on <a href="https://graphviz.org/download/">graphviz.org</a>.
+
+Functions for writing the datatypes in dot are provided. Given the dot code, a command can be used to create a file with the datatype. Make sure that you include the model.mc file. 
+
+	include "path/to/model.mc"
+
+To write the dot code for some data of type `model`, use this command:
+
+	modelPrintDot "YOUR-DATA" "RENDER-DIRECTION"
+
+where "RENDER-DIRECTION" takes one of the following values "TB", "RL", "BT", "LR".
+
+This command then creates your new file:
+
+	dot  -"YOUR-FILETYPE" "NAME-OF-INPUT-FILE" -o "NAME-OF-OUTPUT-FILE"
+
+The filetype decides the type of file you are going to get. It can for example be -Tjpg. -Tps or -Tpdf. The input file will in this case be data-source.js. If you want to take the input directly without a file, the commands can also be piped:
+
+	mi "NAME-OF-CODE-FILE.mc" | dot  -"YOUR-FILETYPE" -o "NAME-OF-OUTPUT-FILE"
+
+# Examples
```

Comment:
The test folder only contains the DFA example, maybe you can put the other example programs there as well.

Reply:
Agree

Reply:
This has been updated! There are now smaller example files in the folder /examples